### ISSUE TITLE
"Invalid char 97 at position 0"

### ISSUE THREAD

**Issue by BudDeuce** [lunarcleint/Wednesdays-Infidelity#16]
### Describe your bug here.

I boot up the game from the default directory, wait for it to load, go into freeplay, select the top week, and attempt to play "Versiculus Iratus." And that is when I get the error message.

I have looked at the .json files of the characters and backgrounds involved and cannot find any problems, I have also run JSON validators online, and they say that the JSON contents are acceptable.

EDIT: I have run every other song and most of the others work fine. I will make a separate issue on another song once this issue is resolved.

### Command Prompt/Terminal logs (if existing)

```bash
Compiling group: haxe
cl.exe -Iinclude -IC:/HaxeToolkit/haxe/lib/discord_rpc/git/linc/ -I C:/HaxeToolkit/haxe/lib/hxCodec/2,5,1/lib/vlc/include/ -I C:/HaxeToolkit/haxe/lib/hxCodec/2,5,1/lib/vlc/src/ -nologo /WX- /fp:precise -DHX_WINDOWS -GR -O2(optim-std) -Zi(debug) -FdC:\Users\ge_ir\Downloads\Wednesdays-Infidelity-master\Wednesdays-Infidelity-master\export\release\windows\obj\obj/msvc1964-nc/vc.pdb(debug) -Od(debug) -O2(release) -Os(optim-size) -FS -Oy- -c -EHs -GS- -IC:/HaxeToolkit/haxe/lib/hxcpp/4,2,1/include -DHXCPP_M64 -DHXCPP_VISIT_ALLOCS(haxe) -DHX_SMART_STRINGS(haxe) -DHXCPP_API_LEVEL=400(haxe) -D_CRT_SECURE_NO_DEPRECATE -D_ALLOW_MSC_VER_MISMATCH -D_ALLOW_ITERATOR_DEBUG_LEVEL_MISMATCH -DHX_WIN_MAIN(main) -wd4996 ... tags=[haxe,static]
 - src/ApplicationMain.cpp
 - src/lime/utils/AssetCache.cpp  [haxe,release]
Link: ApplicationMain.exe
   Creating library ApplicationMain.lib and object ApplicationMain.exp
source/util/Discord.hx:15: Discord Client starting...
source/util/Discord.hx:65: Discord Client initialized
source/util/Discord.hx:22: Discord Client started.
source/data/SpecsDetector.hx:36:
--- SYSTEM INFO ---
MEMORY AMOUNT: 8192
CPU 64 BITS: true
source/song/Conductor.hx:78: new BPM map BUDDY []
```

### Are you modding a build from source or with Lua?

Source

### What is your build target?

Windows

### Did you edit anything in this build? If so, mention or summarize your changes.

Nothing changed.

**Comment by MemeHoovy** [comment]
same issue 🤷🏻‍♂️

**Comment by lunarcleint** [comment]
Does it work in story mode or is it only in freeplay?

**Comment by BudDeuce** [comment]
It does not work in story mode nor freeplay.

**Comment by lunarcleint** [comment]
thats so strange, have you tried resaving the json?

**Comment by BudDeuce** [comment]
I have. I even redownloaded it without looking into the code, just building it—still the same issue.

**Comment by lunarcleint** [comment]
hm what version of windows are you on (it might be a weird haxe thing)

**Comment by BudDeuce** [comment]
I am on Windows 10 21H2 (19044.2075)

**Comment by lunarcleint** [comment]
I wish I knew the issue but im not sure, Ill leave this issue open just incase anyone else wants to add what the problem might be

**Comment by BudDeuce** [comment]
UPDATE: Problems with Versiculus Iratus start with commit e470601. I compared some other songs in commits that had their ram consumption reduced and worked without issue.

**Comment by BudDeuce** [comment]
UPDATE 2: I found the issue, I downloaded the stable version of Wednesday's Infidelity, copied over the "manifest" folder to the game built from the source, and then Versiculus Iratus started to work again.

I had another issue where Leak Ma Balls was crashing as soon as I play it. It now also works as well.

**Comment by MemeHoovy** [comment]
here's the full error from command prompt when loading versiculus iratus
```
source/song/Conductor.hx:78: new BPM map BUDDY []
Called from hxcpp::__hxcpp_main
Called from ApplicationMain::main ApplicationMain.hx line 26
Called from ApplicationMain::create ApplicationMain.hx line 135
Called from lime.app.Application::exec lime/app/Application.hx line 150
Called from lime._internal.backend.native.NativeApplication::exec lime/_internal/backend/native/NativeApplication.hx line 147
Called from lime._internal.backend.native.NativeApplication::handleRenderEvent lime/_internal/backend/native/NativeApplication.hx line 333
Called from lime.app._Event_lime_graphics_RenderContext_Void::dispatch lime/_internal/macros/EventMacro.hx line 91
Called from openfl.display.Stage::__onLimeRender openfl/display/Stage.hx line 2041
Called from openfl.display.Stage::__broadcastEvent openfl/display/Stage.hx line 1175
Called from openfl.display.DisplayObject::__dispatch openfl/display/DisplayObject.hx line 1399
Called from openfl.events.EventDispatcher::__dispatchEvent openfl/events/EventDispatcher.hx line 402
Called from flixel.FlxGame::onEnterFrame flixel/FlxGame.hx line 545
Called from flixel.FlxGame::step flixel/FlxGame.hx line 677
Called from flixel.FlxGame::update flixel/FlxGame.hx line 724
Called from flixel.FlxGame::switchState flixel/FlxGame.hx line 637
Called from states.game.PlayState::create states/game/PlayState.hx line 1071
Called from gameObjects.Character::new gameObjects/Character.hx line 129
Called from flixel.graphics.frames.FlxAtlasFrames::fromTexturePackerJson flixel/graphics/frames/FlxAtlasFrames.hx line 62
Called from haxe.format.JsonParser::doParse C:\HaxeToolkit\haxe\std/haxe/format/JsonParser.hx line 57
Called from haxe.format.JsonParser::parseRec C:\HaxeToolkit\haxe\std/haxe/format/JsonParser.hx line 147
Called from haxe.format.JsonParser::invalidChar C:\HaxeToolkit\haxe\std/haxe/format/JsonParser.hx line 318
```

**Comment by MemeHoovy** [comment]
alright, I found out that "satan-chad" & "sad-demon" both crash the game with the same error whenever you use them in a song

**Comment by lunarcleint** [comment]
interesting, could you send your version of the song json, and character jsons so i can look further into the issue and see whats the problem

**Comment by MemeHoovy** [comment]
here
https://drive.google.com/drive/folders/1vghihwXSFff-XKRAZiVOJeyST6Wox0Bm?usp=sharing

**Comment by Cheemsandfriends** [comment]
char 97 is basically character "a" (you can test this if you have a numeric keypad with alt and then type 97 with a numeric keypad but that's another thing).
unless it's an unvalid json, i believe that youre trying to parse where the json is located, not the actual json lol

**Comment by Cheemsandfriends** [comment]
yknow cos "assets/..."

**Comment by lunarcleint** [comment]
me getting ready to not do anything with that info (im too lazy)

**Comment by lunarcleint** [comment]
![image](https://user-images.githubusercontent.com/88809950/196077253-e3b7bd23-cae4-4419-9a7f-c156e67c3dbc.png)

**Comment by ChandelureF-5** [comment]
h

**Comment by MightyTheArmiddilo** [comment]
Found a different way to fix it, without the manifest folder from the final build. I first looked in the character editor, and checked the two characters VI starts with: mouse-prison and sad-demon. Low and behold: sad-demon crashed with the same error, and I looked deeper, and found that the images/characters/Sad_Demon.json does not match up with the image, images/characters/SAD_DEMON.png, so to fix it, just first: rename Sad_Demon.json to SAD_DEMON.json, then go inside the file and change the image variable in the meta child at the bottom from Sad_Demon.png to SAD_DEMON.png. That's it.

TLDR: Change Sad_Demon.json in the images/characters folder to SAD_DEMON.json and then go inside it and change the image at the bottom of the json to SAD_DEMON.png.

**Comment by BudDeuce** [comment]
Wow! What a throwback. Nice work!

**Comment by BudDeuce** [comment]
> interesting, could you send your version of the song json, and character jsons so i can look further into the issue and see whats the problem

Sure, let me get the files real quick!
